google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/CloudSdkService.java

```diff
@@ -40,7 +44,46 @@ public static CloudSdkService getInstance() {
 
   public abstract void setSdkHomePath(String path);
 
-  public abstract Set<CloudSdkValidationResult> validateCloudSdk(@NotNull Path pathToCloudSdk);
+  protected abstract Set<CloudSdkValidationResult> validateCloudSdk(Path path);
+
+  /**
+   * Checks if the stored path contains a valid Cloud SDK.
+   */
+  public Set<CloudSdkValidationResult> validateCloudSdk() {
```

Comment:
I mentioned it because normally I've seen that we try to keep as little implementation in the abstract service classes as possible, and this seems to be a break from that pattern. @patflynn do you have a strong opinion on whether that's necessary here?